lib/utils/util.js

```diff
@@ -182,3 +182,38 @@ util.encodeU32 = function encodeU32(num) {
   buf[3] = num;
   return buf;
 };
+
+/**
+ * Create a valid mtx with a limit
+ * number of outputs
+ * @param {limit} number
+ * @param {domainOutputMap} Map<string, Array<T>>
+ * @returns { validDomains: Array<String>, invalidDomains: Array<String> }
+ */
+
+util.createBatch = function(limit, domainOutputMap) {
```

Comment:
Broad comment about this method, what if we just do this:

```javascript
const outputs = [];
for (const [name, outputs] of domainOutputMap.entries()) {
  outputs.push(outputs);
}
outputs.sort(/* optional sorting here, eg, sort by output value */);
const selectedOutputs = outputs.slice(0, limit);
return selectedOutputs;
```

Right now this code assumes it's useful to the caller to reveal "all or none of" the bids for a domain, but for our use case specifically, I don't think that's something we need (please lmk if you have reason to believe otherwise). I think it's fine if we reveal an arbitrary subset of the batched outputs, since the rest of our tracking system will continually batch reveal until they're all sent anyway. What do you think?

Btw --^ this also solves the other problem we have of if there's 1 domain with 10000 bids then we can't generate a single transaction for that. I can't think of any problems with only returning an arbitrary subset. 

If you agree that this works --^, then I would also recommend accepting a `sortFunction` so that we can sort reveals by value so we reveal higher value reveals first

Reply:
Hi Anthony,

You are right this method works per domain it reveals all or none. My assumption was like this, a single reveal for a domain sent from a wallet have to reveal all bids (made by the same wallet). I somehow made myself believe we can not partial reveal bids from a wallet. (I didn't see any method to do so, the api provided by the wallet only accepts domain name ) I am failing to see a way to resume from where we leave, considering a scenario where a domain has 201 bids, we revealed the first 200 on the first run, how we will track the remaining last bid for the same domain. 

This is my only concern

Reply:
On the namebase side of things, we'll know whether all of our bids were revealed or not and will try again, and on the handshake side of things, the wallet database knows which bids have already been revealed. It's worth testing if that part works properly since it has not been tested before, but I think the wallet just enumerates all of the bids when deciding what reveals there are to send

Reply:
Hi Anthony,

I did the tests, and yes it's as you have guessed (or as per your comment). It's letting partial bid reveals and sending the next time (after mining a block) reveals the remaining ones.

For the function i do suggest the below one, instead of yours:

`util.createBatch = function(limit, domainOutputMap) {
  if (!limit || limit === 0)
    throw new Error('invalid limit provided!');

  if (!domainOutputMap || domainOutputMap.entries().length === 0) {
      throw new Error('Invalid map provided!');
  }

  const entriesArray = [...domainOutputMap.entries()];
  entriesArray.sort((elem1, elem2) => elem1[1].length > elem2[1].length);

  const rejectedDomains = [];
  const validDomains = [];
  let totalNumberOfOutputs = 0;

  for (const [name, outputs] of entriesArray) {
      const outputLen = totalNumberOfOutputs + outputs.length;
      if (outputLen > limit) {
        // if any free spot left use it
        const availSpots = limit - totalNumberOfOutputs;
        if (availSpots > 0) {
          validDomains.push({name, limit: availSpots});
        }
        // since we failed to process in full, have the name in rejected domains
        // as well, roo
          rejectedDomains.push(name);
      } else {
          totalNumberOfOutputs += outputs.length;
          validDomains.push({name, limit: outputs.length});
      }
  }

  return {validDomains: validDomains, rejectedDomains: rejectedDomains};
};`

Reply:
I am adding new unit tests and will update the PR again.

Reply:
fixed, please check the new function, and let me know if it's ok to proceed this way